### PR TITLE
[Build on demand] Added 'Build now' button to repo page

### LINKED ISSUE #938: Add 'Build now' button to repository page
Design: https://github.com/CanonicalLtd/snappy-design-squad/issues/109

### PR DESCRIPTION
**Pull request by bartaz**
## Done

Added 'Build now' button to repo page (visible only to repository owner).
List of builds is now split between latest builds (on top of the page) and previous page below help boxes.

## QA

- Check out this feature branch
- Run the site using the command ```npm start -- --env=environments/dev.env```
- View the site locally in your web browser at: [http://0.0.0.0:8000/](http://0.0.0.0:8000/)
- Go to repo page
- Builds should be split into latest/previous
- You should be able to trigger builds manually

## Issue / Card

Fixes #938

## Screenshots

screen shot 2017-08-31 at 13 07 19

### CODE REVIEW

**Comment by matthewpaulthomas** [comment]
Until we introduce the daily build settings, it’s not obvious why someone would need to click this button. And if it’s not obvious, there’s a risk that some people may think this is the only way of building their repo.

To avoid this, I suggest putting some text in front of the button to explain its purpose, and using the word “manually” to help distinguish between these builds and the automatic ones. Something like this:

> Has a dependency’s code changed? ( Rebuild Manually Now )

(@evandandrea would be able to confirm whether that text covers what we want it to cover.)

File: src/common/containers/builds.js [hunk 1]
```diff
@@ -2,22 +2,40 @@ import React, { Component, PropTypes } from 'react';
...
+
+  componentWillReceiveProps(nextProps) {
+    // if builds stopped fetching after they were triggered, reset build button status
+    if (this.props.snapBuilds.isFetching && !nextProps.snapBuilds.isFetching) {
+      this.setState({ buildTriggered: false });
+    }
+  }
+
   componentWillMount() {
     const { url } = this.props.repository;
```
[the remark this answers is not in the document]
Comment (cjwatson) [reply src/common/containers/builds.js]: No - that's a [destructuring assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment).  It's the equivalent of `const url = this.props.repository.url;`, but avoids repeating `url` and is less typing to extend if more properties need to be unpacked in future.

**Comment by bartaz** [comment]
@nottrobin Good catch with the badge. I guess we would need to force re-render of the badge to make sure browser requests the badge image once again.

File: src/common/components/badge/index.js [hunk 2]
```diff
@@ -7,14 +7,14 @@ import { CopyToClipboard } from '../share';
 
 import style from './badge.css';
 
-const getBadgeUrl = (fullName) => `${BASE_URL}/badge/${fullName}.svg`;
+const getBadgeUrl = (fullName, buildStatus) => `${BASE_URL}/badge/${fullName}.svg${buildStatus ? '?'+encodeURIComponent(buildStatus) : ''}`;
```
Comment (bartaz) [inline src/common/components/badge/index.js]: Passing current build status to badge URL to force browser to reload the image when build status changes.

**Comment by evandandrea** [comment]
>>Has a dependency’s code changed? ( Rebuild Manually Now )
>(@evandandrea would be able to confirm whether that text covers what we want it to cover.)

I'm fine with that. It doesn't cover the case when you want to rebuild because of a transient failure (network hiccup), but we should keep the message short. Build on dependency change will be far more common (until daily builds).

Weighing into the debate on "Rebuild Manually" vs "Rebuild", I prefer the latter.

**Pull request merged**